design/aws-ecs-priv-registry-support.md

```diff
@@ -14,46 +14,40 @@ The [`ecs.ContainerImage`](https://github.com/awslabs/aws-cdk/blob/master/packag
 
 DockerHub images are assumed public, however DockerHub also provides private repositories and there's currently no way to specify credentials for them in the CDK.
 
-There's also no way to specify images hosted outside of DockerHub, AWS, or your local machine. Customers hosting their own registries or using another registry host, like Quay.io or JFrog Artifactory, would need to be able to specify both the image URI and the registry credentials in order to pull their images down for ECS tasks.
+There's also no explicit way to specify images hosted outside of DockerHub, AWS, or your local machine. Customers hosting their own registries or using another registry host, like Quay.io or JFrog Artifactory, would need to be able to specify both the image URI and the registry credentials in order to pull their images down for ECS tasks.
 
-Therefore, to support the use of private registries we would need to add credentials to the `DockerHubImage` type (as optional), and add a new `PrivateImage` type which could house credentials (required) and the private registry URI.
+Fundamentally, specifying images hosted in DockerHub or elsewhere works the same, because when passed an image URI vs. a plain (or namespaced) image name + tag, the Docker daemon does the right thing and tries to pull the image from the specified registery. 
 
-We should also have a unified way to define repository credentials which can be passed to the constructors of both image types.  
+Therefore, we should rename the existing `DockerHubImage` type be more generic and add the ability to optionally specify credentials.
+ 
 
 ## Code changes
 
-Given the above, we should make the following additions to support private registry authentication:
+Given the above, we should make the following changes to support private registry authentication:
 
 1. Define `RepositoryCredentials` interface & class, add to `IContainerImage` 
-2. Update `DockerHubImage` construct to optionally accept and set `RepositoryCreds` 
-3. Add new `PrivateImage` class which implements `IContainerImage` and requires `RepositoryCreds`
+2. Rename `DockerHubImage` construct to be more generic, optionally accept and set `RepositoryCreds` 
 
 
 # Part 1: How to define registry credentials 
 
-For extensibility, we can define a new `IRepositoryCreds` interface with a "credentialsParamater" string and a new `RepositoryCreds` class which satisfies it using specific constructs (e.g., "fromSecret").
+For extensibility, we can define a new `IRepositoryCreds` interface to house the AWS Secrets Manager secret with the creds and a new `RepositoryCreds` class which satisfies it using specific constructs (e.g., "fromSecret").
 
 ```ts
 export interface IRepositoryCreds {
```

Comment:
Don't call this `IFoo`, just `Foo`. We use interfaces both to specify behavior (`IFoo` has a method `doTheFoo()`) as well as data objects (records/structs/..., `Foo` has a property `fooName: string`).